RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.cc

```diff
@@ -366,22 +372,30 @@ void HGCalCLUEAlgoT<T, STRATEGY>::calculateDistanceToHigher(const T& lt, const u
         // loop over all hits in this bin
         for (unsigned int j = 0; j < binSize; j++) {
           unsigned int otherId = lt[binId][j];
-          float dist = distance(lt, i, otherId, layerId);
+          float dist = distance2(lt, i, otherId, layerId);
           bool foundHigher =
               (cellsOnLayer.rho[otherId] > cellsOnLayer.rho[i]) ||
               (cellsOnLayer.rho[otherId] == cellsOnLayer.rho[i] && cellsOnLayer.detid[otherId] > cellsOnLayer.detid[i]);
-          if (foundHigher && dist <= i_delta) {
-            // update i_delta
+          if (foundHigher && dist < i_delta) {
```

Comment:
wouldn't the following be slightly simpler?
`if (!foundHigher) continue;`

Reply:
I'm not sure I follow entirely your suggestion.

Reply:
Inside a for loop you have:
`if(foundHigher){}
else if(foundHigher){}
else if(foundHigher){}`

It looks like you can simply check if `foundHigher` is false and continue in the for loop if that's the case.
Or do I misread the code?

Reply:
_i.e._ this:
```c++
          if (not foundHigher) {
            continue;
          }
          if (dist < i_delta) {
            rho_max = cellsOnLayer.rho[otherId];
            i_delta = dist;
            i_nearestHigher = otherId;
          } else if (dist == i_delta && cellsOnLayer.rho[otherId] > rho_max) {
            rho_max = cellsOnLayer.rho[otherId];
            i_delta = dist;
            i_nearestHigher = otherId;
          } else if (dist == i_delta && cellsOnLayer.rho[otherId] == rho_max &&
                     cellsOnLayer.detid[otherId] > cellsOnLayer.detid[i]) {
            rho_max = cellsOnLayer.rho[otherId];
            i_delta = dist;
            i_nearestHigher = otherId;
          }
```